doc/Packet-action-DONOTDROP.md

```diff
@@ -0,0 +1,63 @@
+# SAI_PACKET_ACTION_DONOTDROP packet action
+-------------------------------------------------------------------------------
+ Title       | SAI_PACKET_ACTION_DONOTDROP packet action
+-------------|-----------------------------------------------------------------
+ Authors     | Ashish Singh, Google LLC
+ Status      | In review
+ Type        | Standards track
+ Created     | 01/25/2022
+ SAI-Version | 1.8.3
+
+
+-------------------------------------------------------------------------------
+
+This spec discusses the SAI_PACKET_ACTION_DONOTDROP packet action use-cases.
+
+Goal of this packet action attribute (SAI_PACKET_ACTION_DONOTDROP) is to resolve the conflicts of ACL actions when a packet hits more than one ACLs such that when this packet action is used with an ACL, it serves as an action that prevents the possible drop from another ACL that is in a lower priority table.
+
+This new packet action attribute is in line with the current SAI architecture - actions in higher priority groups take precedence. If an ACL with DROP action is in a higher priority group than this ACL then, DROP action is expected to be honored. Few cases are explained with examples in next section.
+
+# Usage Examples
+### Case 1
+
+| ACL in table T1                     | ACL in table T2                       |
+|-------------------------------------|---------------------------------------|
+| Table priority group:  P1           | Table priority group:  P2             |
+|Action: SAI_PACKET_ACTION_DONOTDROP  | Action: DROP                          |
+
+
+
+If P1 > P2: action is “no DROP”
+
+
+If P2 > P1: action is “DROP”
+
+
+
+### Case 2
+
+| ACL in table T1                     | ACL in table T2                       |
+|-------------------------------------|---------------------------------------|
+| Table priority group:  P1           | Table priority group:  P2             |
+|Action: SAI_PACKET_ACTION_DONOTDROP  | Action: neither DROP nor SAI_PACKET_ACTION_DONOTDROP                         |
+
+
+If P1 > P2: action is “no DROP”
+
+
+If P2 > P1: action is “no DROP”
+
+
+### Case 3
+
+| ACL in table T1                     | ACL in table T2                       |
+|-------------------------------------|---------------------------------------|
+| Table priority group:  P1           | Table priority group:  P2             |
+|Action: neither DROP nor SAI_PACKET_ACTION_DONOTDROP  | Action: DROP                         |
+
+
```

Comment:
We need to define the behavior for enums that are "Combination of Packet Actions" like  SAI_PACKET_ACTION_TRAP. For example, can you explain the below:

| ACL in table T1                     | ACL in table T2                       |
|-------------------------------------|---------------------------------------|
| Table priority group:  P1           | Table priority group:  P2             |
|Action: SAI_PACKET_ACTION_DONOTDROP  | Action: TRAP                         |

If P1 > P2: action is “Copy to CPU?”


If P2 > P1: action is “Trap?”

Reply:
Added behavior of DONOTDROP with TRAP and DENY. Thank you.

Reply:
@rck-innovium would you take a look at whether the documentation addresses your question? Thank you.